### PR TITLE
Tilemanager testing

### PR DESCRIPTION
**Pull request by hjanetzek**

### CODE REVIEW

File: core/src/tile/tileManager.cpp [hunk 1]
```diff
@@ -473,11 +460,12 @@ void TileManager::updateProxyTiles(TileSet& _tileSet, const TileID& _tileID, Til
         return;
     }
     // Try children
-    if (m_view->s_maxZoom > _tileID.z) {
-        for (int i = 0; i < 4; i++) {
-            updateProxyTile(_tileSet, _tile, _tileID.getChild(i), static_cast<ProxyID>(1 << i));
-        }
+    // FIXME maxZoom also depends on DataSource
```
Comment (tallytalwar) [inline core/src/tile/tileManager.cpp]: Yup, good catch, view maxZoom and dataSource maxZoom can be different. A legit FIXME :D.

File: tests/unit/tileManagerTests.cpp [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+    tileManager.updateTileSets(viewState, visibleTiles);
+
+    REQUIRE(tileManager.getVisibleTiles().size() == 1);
+    REQUIRE(source->tileTaskCount == 1);
+    REQUIRE(worker.processedCount == 1);
+
+}
+
+
+TEST_CASE( "Use proxy Tile", "[TileManager][updateTileSets]" ) {
```
Comment (tallytalwar) [inline tests/unit/tileManagerTests.cpp]: Can we add a test case for use of grand parent proxy?

Comment (hjanetzek) [inline tests/unit/tileManagerTests.cpp]: Sure! This is just a start to make TileManager testable. Optimally we should add tests for all the logic we currently expect and then for every change we should first add some tests.

Comment (tallytalwar) [inline tests/unit/tileManagerTests.cpp]: Agreed!

**Comment by tallytalwar** [comment]
I am good with this also. Ofcourse we can have more tests for tilemanager functionality,  but this is a start. Once we are all green from Travis, can merge this. @blair1618 you should also have a look at this. 

Thanks

**Comment by hjanetzek** [comment]
The test already already catched a regression in https://github.com/tangrams/tangram-es/commit/b3d7c8996519607bf4778cac6988675e28a34c04

I'm working on it

**Pull request closed without merging**